### PR TITLE
Pv2 270 v2 two levy learners one employer but two providers and not enough levy

### PR DESCRIPTION
**Pull request by samuel-dev-dba**

### CODE REVIEW

File: src/SFA.DAS.Payments.AcceptanceTests.EndToEnd/Steps/EndToEndStepsBase.cs [hunk 1]
```diff
@@ -825,24 +831,43 @@ protected async Task GeneratedAndValidateEarnings(Table table, Provider provider
...
             await WaitForIt(() => matcher.MatchPayments(), "Earning event check failure");
         }
-        
+
         protected async Task HandleIlrReSubmissionForTheLearners(string collectionPeriodText, Provider provider)
         {
             var collectionPeriod = new CollectionPeriodBuilder().WithSpecDate(collectionPeriodText).Build();
-            if (Context.ContainsKey("current_collection_period") && (CurrentCollectionPeriod.Period != collectionPeriod.Period || CurrentCollectionPeriod.AcademicYear != collectionPeriod.AcademicYear))
+
+            if (ProvidersWithCacheCleared == null)
```
Comment (kuzinator) [inline src/SFA.DAS.Payments.AcceptanceTests.EndToEnd/Steps/EndToEndStepsBase.cs]: This looks like twice two times the same

File: src/SFA.DAS.Payments.AcceptanceTests.EndToEnd/Steps/EndToEndStepsBase.cs [hunk 2]
```diff
@@ -825,24 +831,43 @@ protected async Task GeneratedAndValidateEarnings(Table table, Provider provider
...
+                    ProvidersWithCacheCleared = new Dictionary<(byte period, int academicYear, long ukprn), long>
+                    {
+                        {(collectionPeriod.Period, collectionPeriod.AcademicYear, provider.Ukprn), provider.Ukprn }
+                    };
+
+                SetCollectionPeriod(collectionPeriodText);
+                return;
+            }
+
+            if (!ProvidersWithCacheCleared.ContainsKey((collectionPeriod.Period, collectionPeriod.AcademicYear, provider.Ukprn)))
```
Comment (kuzinator) [inline src/SFA.DAS.Payments.AcceptanceTests.EndToEnd/Steps/EndToEndStepsBase.cs]: This needs to be invoked only on subsequent runs, not on the first one

Comment (samuel-dev-dba) [reply src/SFA.DAS.Payments.AcceptanceTests.EndToEnd/Steps/EndToEndStepsBase.cs]: changed

File: src/SFA.DAS.Payments.AcceptanceTests.EndToEnd/Steps/EndToEndStepsBase.cs (hunk 1 above)
Comment (samuel-dev-dba) [reply src/SFA.DAS.Payments.AcceptanceTests.EndToEnd/Steps/EndToEndStepsBase.cs]: changed

File: src/SFA.DAS.Payments.AcceptanceTests.EndToEnd/Steps/EndToEndStepsBase.cs (hunk 2 above)
Comment (kuzinator) [reply src/SFA.DAS.Payments.AcceptanceTests.EndToEnd/Steps/EndToEndStepsBase.cs]: Minor point, you declare this as a dictionary but never use the value. Is this a case for a hash set?

**Pull request merged**